client/app/pages/queries/components/QueryPageHeader.jsx

```diff
@@ -53,133 +59,126 @@ function createMenu(menu) {
   );
 }
 
-export default function QueryPageHeader({ query, sourceMode, onChange }) {
-  const saveName = useCallback(
-    name => {
-      renameQuery(query, name).then(onChange);
-    },
-    [query, onChange]
-  );
+export default function QueryPageHeader({ query, dataSource, sourceMode, selectedVisualization, onChange }) {
+  const queryFlags = useQueryFlags(query, dataSource);
 
-  const saveTags = useCallback(
-    tags => {
-      updateQuery(query, { tags }).then(onChange);
-    },
-    [query, onChange]
-  );
+  const updateName = useRenameQuery(query, onChange);
+
```

Comment:
I prefer it without that many `\n`, but up to you :slightly_smiling_face:

Reply:
Fixed :slightly_smiling_face: